snapcraft/internal/meta.py

```diff
@@ -325,7 +325,9 @@ def _reformat_desktop(self, name, desktop_file):
                 'The specified desktop file {!r} is missing the '
                 '"Exec" key'.format(desktop_file))
         # XXX: do we want to allow more parameters for Exec?
-        exec_value = '{}.{} %U'.format(self._config_data['name'], name)
+        snap_name = self._config_data['name']
+        exec_value = '{} %U'.format(
+            name if name == snap_name else '{}.{}'.format(snap_name, name))
```

Comment:
The nested if/else and format cause a double-take. I good old-fashioned if/else would be easier to read.